modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/helper/AddFormInstanceRecordMVCCommandHelper.java

```diff
@@ -66,229 +53,145 @@
 public class AddFormInstanceRecordMVCCommandHelper {
 
 	public void updateNonevaluableDDMFormFields(
-			ActionRequest actionRequest, DDMForm ddmForm,
-			DDMFormValues ddmFormValues, Locale locale)
+			Map<String, DDMFormField> ddmFormFieldsMap,
+			Map<DDMFormEvaluatorFieldContextKey, Map<String, Object>>
+				ddmFormFieldsPropertyChanges,
+			Map<String, List<DDMFormFieldValue>> ddmFormFieldValuesMap,
+			DDMFormLayout ddmFormLayout, Set<Integer> disabledPagesIndexes)
 		throws Exception {
 
-		DDMFormEvaluatorEvaluateResponse ddmFormEvaluatorEvaluateResponse =
-			_evaluate(actionRequest, ddmForm, ddmFormValues, locale);
-
-		Set<String> nonevaluableFieldNames = _getNonevaluableFieldNames(
-			ddmFormEvaluatorEvaluateResponse);
-
-		nonevaluableFieldNames.addAll(
-			_getFieldNamesFromDisabledPages(
-				_getDDMFormLayout(actionRequest),
-				ddmFormEvaluatorEvaluateResponse.getDisabledPagesIndexes()));
+		Set<String> nonevaluableFieldNames = new HashSet<>();
 
-		_removeDDMFormFieldValues(
-			ddmFormValues.getDDMFormFieldValuesMap(true),
-			nonevaluableFieldNames);
-		_updateNonevaluableDDMFormFields(
-			ddmForm.getDDMFormFieldsMap(true), nonevaluableFieldNames);
-	}
+		for (Map.Entry<DDMFormEvaluatorFieldContextKey, Map<String, Object>>
+				entry : ddmFormFieldsPropertyChanges.entrySet()) {
 
-	public void validateExpirationStatus(
-			DDMFormInstance ddmFormInstance, PortletRequest portletRequest)
-		throws Exception {
+			if (!MapUtil.getBoolean(entry.getValue(), "readOnly") &&
+				MapUtil.getBoolean(entry.getValue(), "visible", true)) {
```

Comment:
@gabrielwas I'm not sure I understand what you're proposing.

We don't need to call the three parameters for `MapUtil.getBoolean(entry.getValue(), "readOnly")` because, if the map does not have the `readOnly` key, the default value should be `false`.  It would be redundant to do `MapUtil.getBoolean(entry.getValue(), "readOnly", false)` because of this code [here](https://github.com/liferay/liferay-portal/blob/1c711d37b175890e02a0f6f2f7345592fe984110/portal-kernel/src/com/liferay/portal/kernel/util/MapUtil.java#L65-L70).

We **need** to call the three parameter method for `MapUtil.getBoolean(entry.getValue(), "visible", true)` because, if the `visible` property does not exist on this map (meaning that no fields were affected by a rule to hide it), then the default value for `visible` must be true.

Reply:
Got it. I just think it's weird using different methods for the same purpose. But you're right. It's ok like that.